pyccel/codegen/printing/ccode.py

```diff
@@ -778,6 +778,43 @@ def _print_MathSqrt(self, expr):
             code_args = self._print(arg)
         return 'sqrt({})'.format(code_args)
 
+    # -------------- pyc_math functions ---------------------------------------
+    def _print_MathFactorial(self, expr):
+        """ Print the equivalent code to pycthon factorial function call"""
+        # include of the pyc_math header
+        self._additional_imports.add('pyc_math')
+        arg = expr.args[0]
+        if arg.dtype is not NativeInteger() or len(expr.args) > 1:
+            raise ("factoral function accept one integer argument")
```

Comment:
Can you explain why this is a C specific problem? Looking at the docstring for `math.factorial` it seems that this rule is true in the original python, so it should not matter which language we are printing in. In that case this kind of check should be done in the `MathFactorial` constructor

Reply:
I think this check should not be in the printer because it's a common error for both languages c and Fortran.
and these errors will be raised in python checks before using pyccel.